Azure-Pipelines: update *.yml file and build scripts

amiclaus opened the pull request:
With this PR, the Azure-Pipelines integration is done.
Build jobs:
- code style check
- static analysis
- m1k-fw build

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

amiclaus committed:
azure: update pipeline.yml

Add custom content for the .yml file.

The jobs for this repository are performing code style check, static
analysis and m1k firmware build.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

amiclaus committed:
lib.sh: update paths

Update build environment variables.

Create new env var that replaces the TravisCI specific one.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

amiclaus committed:
run_build: add .sh file extension

Add proper file extension for the script.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

commodo commented:
`run_build: add .sh file extension`  - the rename extension commit looks a bit redundant; if doing it, might as well go full-out and move eveything from `ci/travis/ to `ci`
it's just a thought; we can leave it as-is

in any case, cppcheck has some errors;
we should fix them before merging
```
asf/common/services/usb/class/vendor/device/udi_vendor_desc.c:65:1: error: There is an unknown macro here somewhere. Configuration is required. If UDC_DATA is a macro then please configure it. [unknownMacro]
UDC_DATA(4)
^
asf/sam/utils/cmsis/sam3u/source/templates/gcc/startup_sam3u.c:149:30: error: Comparing pointers that point to different objects [comparePointers]
 for (pDest = &_szero; pDest < &_ezero;) {
                             ^
asf/sam/utils/cmsis/sam3u/source/templates/gcc/startup_sam3u.c:54:17: note: Variable declared here.
extern uint32_t _szero;
                ^
asf/sam/utils/cmsis/sam3u/source/templates/gcc/startup_sam3u.c:149:15: note: Address of variable taken here.
 for (pDest = &_szero; pDest < &_ezero;) {
              ^
asf/sam/utils/cmsis/sam3u/source/templates/gcc/startup_sam3u.c:55:17: note: Variable declared here.
extern uint32_t _ezero;
                ^
asf/sam/utils/cmsis/sam3u/source/templates/gcc/startup_sam3u.c:149:32: note: Address of variable taken here.
 for (pDest = &_szero; pDest < &_ezero;) {
                               ^
asf/sam/utils/cmsis/sam3u/source/templates/gcc/startup_sam3u.c:149:30: note: Comparing pointers that point to different objects
 for (pDest = &_szero; pDest < &_ezero;) {

```

amiclaus committed:
ci: add ignore files

Add `.cppcheckignore` and `.astyleignore` files.

Third party software components from this repository are ignored from
code style check and static analysis.

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

amiclaus commented:
> 
> 
> `run_build: add .sh file extension` - the rename extension commit looks a bit redundant; if doing it, might as well go full-out and move eveything from `ci/travis/ to `ci`
> it's just a thought; we can leave it as-is
> 
> in any case, cppcheck has some errors;
> we should fix them before merging
> 
> ```
> asf/common/services/usb/class/vendor/device/udi_vendor_desc.c:65:1: error: There is an unknown macro here somewhere. Configuration is required. If UDC_DATA is a macro then please configure it. [unknownMacro]
> UDC_DATA(4)
> ^
> asf/sam/utils/cmsis/sam3u/source/templates/gcc/startup_sam3u.c:149:30: error: Comparing pointers that point to different objects [comparePointers]
>  for (pDest = &_szero; pDest < &_ezero;) {
>                              ^
> asf/sam/utils/cmsis/sam3u/source/templates/gcc/startup_sam3u.c:54:17: note: Variable declared here.
> extern uint32_t _szero;
>                 ^
> asf/sam/utils/cmsis/sam3u/source/templates/gcc/startup_sam3u.c:149:15: note: Address of variable taken here.
>  for (pDest = &_szero; pDest < &_ezero;) {
>               ^
> asf/sam/utils/cmsis/sam3u/source/templates/gcc/startup_sam3u.c:55:17: note: Variable declared here.
> extern uint32_t _ezero;
>                 ^
> asf/sam/utils/cmsis/sam3u/source/templates/gcc/startup_sam3u.c:149:32: note: Address of variable taken here.
>  for (pDest = &_szero; pDest < &_ezero;) {
>                                ^
> asf/sam/utils/cmsis/sam3u/source/templates/gcc/startup_sam3u.c:149:30: note: Comparing pointers that point to different objects
>  for (pDest = &_szero; pDest < &_ezero;) {
> ```

The cppcheck issues are detected on the third-party software components from this repository.
Fixed with: https://github.com/analogdevicesinc/m1k-fw/pull/30/commits/6cbb97e85b377081d4ec4c692a2638ccaeb68268
